react/src/components/KeypairResourcePolicySettingModal.tsx

```diff
@@ -245,19 +249,20 @@ const KeypairResourcePolicySettingModal: React.FC<
               props: props as ModifyKeyPairResourcePolicyInput,
             },
             onCompleted: (res, errors) => {
-              if (!res?.modify_keypair_resource_policy?.ok) {
-                message.error(res?.modify_keypair_resource_policy?.msg);
-                onRequestClose();
-                return;
-              }
-              if (errors && errors?.length > 0) {
-                const errorMsgList = _.map(errors, (error) => error.message);
-                for (const error of errorMsgList) {
-                  message.error(error, 2.5);
+              if (!res?.modify_keypair_resource_policy?.ok || errors) {
+                if (res.modify_keypair_resource_policy?.msg) {
+                  message.error(
+                    painKiller.relieve(res.modify_keypair_resource_policy.msg),
+                  );
+                } else if (errors && errors.length > 0) {
+                  // if res.create_keypair_resource_policy is null
+                  errors.forEach((error) =>
+                    message.error(painKiller.relieve(error.message), 2.5),
+                  );
                 }
-                onRequestClose();
+                onRequestClose(false);
```

Comment:
It looks like the condition for the if statement is set incorrectly. And if you have multiple nested conditions in an if statement, an if-return statement may be more readable than an if-else statement. This is a code styling difference, so just give it some thought.

```suggestion
              if (
                !res?.create_keypair_resource_policy?.ok &&
                res.create_keypair_resource_policy?.msg
              ) {
                message.error(
                  painKiller.relieve(res.create_keypair_resource_policy.msg),
                );
                return;
              }

              if (errors && errors.length > 0) {
                errors.forEach((error) =>
                  message.error(painKiller.relieve(error.message), 2.5),
                );
                return;
              }
              message.success(t('resourcePolicy.SuccessfullyCreated'));
              onRequestClose(true);
```

Reply:
It was. Thank you!